python/fledge/common/logger.py

```diff
@@ -181,10 +202,37 @@ def get_logger(self, logger_name: str):
             logger: returns logger for module
         """
         _logger = logging.getLogger(logger_name)
+
+        # save the old logging.error function
+        __logging_error = _logger.error
+
         console_handler = self.get_console_handler()
         syslog_handler = self.get_syslog_handler()
         self.add_handlers(_logger, [syslog_handler, console_handler])
         _logger.propagate = False
+
+        @wraps(_logger.error)
+        def error(msg, *args, **kwargs):
+            """Case: When we pass exception in error and having different args
+            For example:
+            a) _logger.error(ex)
+            b) _logger.error(ex, "Failed to add data.")
+            """
```

Comment:
""" Override error logger to print multi-line traceback and error string with newline"

case/example part can be moved to if block, as given in the else part.